nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/UnpackContent.java

```diff
@@ -394,6 +412,65 @@ public void process(final OutputStream out) throws IOException {
         }
     }
 
+    private static class SevenZipUnpacker extends Unpacker {
+        public SevenZipUnpacker(Pattern fileFilter) {
+            super(fileFilter);
+        }
+
+        @Override
+        public void unpack(final ProcessSession session, final FlowFile source, final List<FlowFile> unpacked) {
+            final String fragmentId = UUID.randomUUID().toString();
+            session.read(source, new InputStreamCallback() {
+                @Override
+                public void process(final InputStream in) throws IOException {
+
+                    int fragmentCount = 0;
+                    byte[] inputData = IOUtils.toByteArray(in);
```

Comment:
This could easily cause the JVM to run out of memory. It looks like what you might need to do is create a temporary `java.io.File` and export the contents of the flowfile to that because the SevenZFile class doesn't allow you to natively use an InputStream. That could be dangerous in its own right, but at least users could mitigate it by resizing the partition if there is sufficient documentation warning them.

Reply:
Sounds good - is there a standard place in the NiFi structure to put temporary files or do I let the user specify with a new property? If it's with a new property, is there another processor that does this so I can align the name and description?

Reply:
I'd have to think about this. I saw an example on google of converting an InputStream into the seekablesource (or whatever it's called), but it didn't say whether or not the process put it all in memory. We should probably take a little time to run that down to see if it's possible to safely wrap an InputStream before trying a temp file.